### PR TITLE
[lldb] Skip assert of type validation for imported types

### PR DESCRIPTION
**Pull request by augusto2112**
Skip assertions comparing the type constructed by TypeSystemSwiftTypeRef
and SwiftASTContext for imported types, since often the two type systems
will have different, but not necessarily incorrect, views of the same
type. At the same time we keep the llvm::dbgs logging in case we're
actually dealing with a problematic imported type.

rdar://96166324

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by JDevlieghere** [comment]
Thanks Augusto!

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -1862,15 +1862,19 @@ constexpr ExecutionContextScope *g_no_exe_ctx = nullptr;
...
     if (!equivalent)                                                           \
       llvm::dbgs() << "failing type was " << (const char *)TYPE << "\n";       \
-    assert(equivalent &&                                                       \
-           "TypeSystemSwiftTypeRef diverges from SwiftASTContext");            \
+    /* Only assert if it's not an imported type. The typeref type system and   \
+    the ast context often have very different views of the same type           \
+     without there being anything nece wrong. */                               \
+    if (!IsImportedType(TYPE, nullptr))                                        \
+      assert(equivalent &&                                                     \
+             "TypeSystemSwiftTypeRef diverges from SwiftASTContext");          \
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1877]: I think this is fine if temporary, but should it be permanent? Do we know that this doesn't silence some real issues?

Comment (augusto2112) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1877]: We still have the llvm::dbgs right before this, so if some test starts failing we still have some signal of what the problem might be. If you think it's too broad we can add the failing type to `ShouldSkipValidation` instead.

Comment (kastiglione) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1877]: I am worried we wouldn't see the debug output. Also, our tests are far from exhaustive.

A higher level question is how long we plan to keep the assertions. My understanding is they were added to ensure the transition to TypeSystemSwiftTypeRef. Maybe the time will soon come that we delete these assertions, in which case maybe this is fine. However we could argue that we'd be less confident to remove the assertions given that we've weakened this one without fully knowing.

Comment (augusto2112) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:1877]: Ok here's a patch skipping validation for this specific type https://github.com/apple/llvm-project/pull/4915

**Pull request closed without merging**